crates/celo-alloy/consensus/src/transaction/cip64.rs

```
@@ -177,7 +180,7 @@ impl RlpEcdsaDecodableTx for TxCip64 {
             value: Decodable::decode(buf)?,
             input: Decodable::decode(buf)?,
             access_list: Decodable::decode(buf)?,
-            fee_currency: Decodable::decode(buf)?,
+            fee_currency: Decodable::decode(buf).ok(),
```

We can know if the fee_currency is unset without swallowing all errors (I think buf will be empty and the probably the RLP or list header would also indicate this). This kind of edge case error handling is not urgent, but we should at  least leave a TODO here.

---

Handled in https://github.com/celo-org/celo-kona/pull/43/commits/be88c15ae2b6fcf541d354fb226310317e16d3a8